lib/configs/_commons.js

```diff
@@ -1,22 +1,26 @@
 "use strict"
 
-/**
- * @type {import('eslint').Linter.RulesRecord}
- */
-module.exports.commonRules = {
-    "n/no-deprecated-api": "error",
-    "n/no-extraneous-import": "error",
-    "n/no-extraneous-require": "error",
-    "n/no-exports-assign": "error",
-    "n/no-missing-import": "error",
-    "n/no-missing-require": "error",
-    "n/no-process-exit": "error",
-    "n/no-unpublished-bin": "error",
-    "n/no-unpublished-import": "error",
-    "n/no-unpublished-require": "error",
-    "n/no-unsupported-features/es-builtins": "error",
-    "n/no-unsupported-features/es-syntax": "error",
-    "n/no-unsupported-features/node-builtins": "error",
-    "n/process-exit-as-throw": "error",
-    "n/hashbang": "error",
-}
+const rules = require("../all-rules")
+
+/** @type {import('eslint').Linter.RulesRecord} */
+const recommendeRulesConfig = {}
+
+/** @type {import('eslint').Linter.RulesRecord} */
+const allRulesConfig = {}
+
+Object.keys(rules).map(ruleName => {
+    const scopedRuleName = `n/${ruleName}`
+    const rule = rules[ruleName]
+
+    // only add rules that are not deprecated
+    if (rule?.meta?.deprecated !== true) {
```

Comment:
the `?.` is only to make `tsc` happy. 😂

Reply:
You can try using `Object.entries` instead which removes the need to the look and makes the compiler happy.

eg. `for(const [ruleName, rule] of Object.entries(rules) { ... }`

Reply:
good call!👍